### ISSUE TITLE
Utilize our NG ruleset file for code analysis

### ISSUE THREAD

**Issue by bbarber** [dontpaniclabs/SolutionTemplate#11]
I expected each `.csproj` file to reference our code analysis ruleset file. I'd also expect 
`<TreatWarningsAsErrors>true</TreatWarningsAsErrors>`

Was this intentional to not use this code analysis ruleset? @twilsman

**Comment by bbarber** [comment]
If this is desired, I would nominate @tarrball to include a ruleset file. He has created his own preferred ruleset on tenzing which probably is a good overlap to our original NG ruleset file.

**Comment by bbarber** [comment]
Merged ruleset (tarr + NG)

```
<?xml version="1.0" encoding="utf-8"?>
<RuleSet Name="NgRules" ToolsVersion="16.0">
  <Include Path="basiccorrectnessrules.ruleset" Action="Default" />
  <Include Path="basicdesignguidelinerules.ruleset" Action="Default" />
  <Include Path="extendedcorrectnessrules.ruleset" Action="Default" />
  <Include Path="minimumrecommendedrules.ruleset" Action="Default" />
  <Rules AnalyzerId="Microsoft.Analyzers.ManagedCodeAnalysis" RuleNamespace="Microsoft.Rules.Managed">
    <Rule Id="CA1000" Action="None" />
    <Rule Id="CA1001" Action="Error" />
    <Rule Id="CA1002" Action="None" />
    <Rule Id="CA1003" Action="Error" />
    <Rule Id="CA1004" Action="None" />
    <Rule Id="CA1005" Action="Error" />
    <Rule Id="CA1006" Action="None" />
    <Rule Id="CA1007" Action="Error" />
    <Rule Id="CA1008" Action="Error" />
    <Rule Id="CA1009" Action="Error" />
    <Rule Id="CA1010" Action="Error" />
    <Rule Id="CA1011" Action="None" />
    <Rule Id="CA1012" Action="Error" />
    <Rule Id="CA1013" Action="Error" />
    <Rule Id="CA1014" Action="None" />
    <Rule Id="CA1016" Action="Error" />
    <Rule Id="CA1017" Action="Error" />
    <Rule Id="CA1018" Action="Error" />
    <Rule Id="CA1019" Action="Error" />
    <Rule Id="CA1023" Action="Error" />
    <Rule Id="CA1024" Action="None" />
    <Rule Id="CA1025" Action="Error" />
    <Rule Id="CA1026" Action="None" />
    <Rule Id="CA1027" Action="Error" />
    <Rule Id="CA1028" Action="Error" />
    <Rule Id="CA1030" Action="Error" />
    <Rule Id="CA1031" Action="None" />
    <Rule Id="CA1032" Action="Error" />
    <Rule Id="CA1033" Action="Error" />
    <Rule Id="CA1034" Action="Error" />
    <Rule Id="CA1035" Action="Error" />
    <Rule Id="CA1036" Action="Error" />
    <Rule Id="CA1038" Action="Error" />
    <Rule Id="CA1039" Action="Error" />
    <Rule Id="CA1041" Action="Error" />
    <Rule Id="CA1043" Action="Error" />
    <Rule Id="CA1044" Action="Error" />
    <Rule Id="CA1046" Action="None" />
    <Rule Id="CA1047" Action="Error" />
    <Rule Id="CA1048" Action="Error" />
    <Rule Id="CA1049" Action="Error" />
    <Rule Id="CA1050" Action="Error" />
    <Rule Id="CA1051" Action="Error" />
    <Rule Id="CA1052" Action="None" />
    <Rule Id="CA1053" Action="Error" />
    <Rule Id="CA1054" Action="Error" />
    <Rule Id="CA1055" Action="Error" />
    <Rule Id="CA1056" Action="None" />
    <Rule Id="CA1057" Action="Error" />
    <Rule Id="CA1058" Action="Error" />
    <Rule Id="CA1059" Action="Error" />
    <Rule Id="CA1060" Action="Error" />
    <Rule Id="CA1061" Action="Error" />
    <Rule Id="CA1063" Action="Error" />
    <Rule Id="CA1064" Action="Error" />
    <Rule Id="CA1065" Action="Error" />
    <Rule Id="CA1301" Action="Error" />
    <Rule Id="CA1303" Action="None" />
    <Rule Id="CA1308" Action="Error" />
    <Rule Id="CA1400" Action="Error" />
    <Rule Id="CA1401" Action="Error" />
    <Rule Id="CA1402" Action="Error" />
    <Rule Id="CA1403" Action="Error" />
    <Rule Id="CA1404" Action="Error" />
    <Rule Id="CA1405" Action="Error" />
    <Rule Id="CA1406" Action="Error" />
    <Rule Id="CA1407" Action="Error" />
    <Rule Id="CA1408" Action="Error" />
    <Rule Id="CA1409" Action="Error" />
    <Rule Id="CA1410" Action="Error" />
    <Rule Id="CA1411" Action="Error" />
    <Rule Id="CA1412" Action="Error" />
    <Rule Id="CA1413" Action="None" />
    <Rule Id="CA1414" Action="Error" />
    <Rule Id="CA1415" Action="Error" />
    <Rule Id="CA1500" Action="Error" />
    <Rule Id="CA1502" Action="Error" />
    <Rule Id="CA1600" Action="Error" />
    <Rule Id="CA1601" Action="Error" />
    <Rule Id="CA1708" Action="Error" />
    <Rule Id="CA1716" Action="None" />
    <Rule Id="CA1801" Action="Error" />
    <Rule Id="CA1804" Action="Error" />
    <Rule Id="CA1806" Action="Error" />
    <Rule Id="CA1809" Action="Error" />
    <Rule Id="CA1810" Action="Error" />
    <Rule Id="CA1811" Action="Error" />
    <Rule Id="CA1812" Action="Error" />
    <Rule Id="CA1813" Action="Error" />
    <Rule Id="CA1814" Action="None" />
    <Rule Id="CA1815" Action="None" />
    <Rule Id="CA1816" Action="Error" />
    <Rule Id="CA1819" Action="None" />
    <Rule Id="CA1820" Action="Error" />
    <Rule Id="CA1821" Action="Error" />
    <Rule Id="CA1822" Action="None" />
    <Rule Id="CA1823" Action="Error" />
    <Rule Id="CA1824" Action="Error" />
    <Rule Id="CA1900" Action="Error" />
    <Rule Id="CA1901" Action="Error" />
    <Rule Id="CA1903" Action="Error" />
    <Rule Id="CA2001" Action="Error" />
    <Rule Id="CA2002" Action="Error" />
    <Rule Id="CA2003" Action="Error" />
    <Rule Id="CA2004" Action="Error" />
    <Rule Id="CA2006" Action="Error" />
    <Rule Id="CA2007" Action="None" />
    <Rule Id="CA2101" Action="Error" />
    <Rule Id="CA2102" Action="Error" />
    <Rule Id="CA2104" Action="Error" />
    <Rule Id="CA2105" Action="Error" />
    <Rule Id="CA2106" Action="Error" />
    <Rule Id="CA2108" Action="Error" />
    <Rule Id="CA2111" Action="Error" />
    <Rule Id="CA2112" Action="Error" />
    <Rule Id="CA2114" Action="Error" />
    <Rule Id="CA2115" Action="Error" />
    <Rule Id="CA2116" Action="Error" />
    <Rule Id="CA2117" Action="Error" />
    <Rule Id="CA2119" Action="Error" />
    <Rule Id="CA2120" Action="Error" />
    <Rule Id="CA2121" Action="Error" />
    <Rule Id="CA2122" Action="Error" />
    <Rule Id="CA2123" Action="Error" />
    <Rule Id="CA2124" Action="Error" />
    <Rule Id="CA2126" Action="Error" />
    <Rule Id="CA2130" Action="Error" />
    <Rule Id="CA2131" Action="Error" />
    <Rule Id="CA2132" Action="Error" />
    <Rule Id="CA2133" Action="Error" />
    <Rule Id="CA2134" Action="Error" />
    <Rule Id="CA2135" Action="Error" />
    <Rule Id="CA2136" Action="Error" />
    <Rule Id="CA2137" Action="Error" />
    <Rule Id="CA2138" Action="Error" />
    <Rule Id="CA2139" Action="Error" />
    <Rule Id="CA2140" Action="Error" />
    <Rule Id="CA2141" Action="Error" />
    <Rule Id="CA2142" Action="Error" />
    <Rule Id="CA2143" Action="Error" />
    <Rule Id="CA2144" Action="Error" />
    <Rule Id="CA2145" Action="Error" />
    <Rule Id="CA2146" Action="Error" />
    <Rule Id="CA2147" Action="Error" />
    <Rule Id="CA2149" Action="Error" />
    <Rule Id="CA2200" Action="Error" />
    <Rule Id="CA2201" Action="Error" />
    <Rule Id="CA2202" Action="Error" />
    <Rule Id="CA2204" Action="None" />
    <Rule Id="CA2205" Action="Error" />
    <Rule Id="CA2207" Action="Error" />
    <Rule Id="CA2208" Action="Error" />
    <Rule Id="CA2211" Action="Error" />
    <Rule Id="CA2212" Action="Error" />
    <Rule Id="CA2213" Action="None" />
    <Rule Id="CA2214" Action="Error" />
    <Rule Id="CA2215" Action="Error" />
    <Rule Id="CA2216" Action="Error" />
    <Rule Id="CA2217" Action="Error" />
    <Rule Id="CA2218" Action="Error" />
    <Rule Id="CA2219" Action="Error" />
    <Rule Id="CA2220" Action="Error" />
    <Rule Id="CA2221" Action="Error" />
    <Rule Id="CA2222" Action="Error" />
    <Rule Id="CA2223" Action="Error" />
    <Rule Id="CA2224" Action="Error" />
    <Rule Id="CA2225" Action="None" />
    <Rule Id="CA2226" Action="Error" />
    <Rule Id="CA2227" Action="None" />
    <Rule Id="CA2228" Action="Error" />
    <Rule Id="CA2229" Action="Error" />
    <Rule Id="CA2230" Action="Error" />
    <Rule Id="CA2231" Action="Error" />
    <Rule Id="CA2232" Action="Error" />
    <Rule Id="CA2233" Action="Error" />
    <Rule Id="CA2234" Action="Error" />
    <Rule Id="CA2235" Action="Error" />
    <Rule Id="CA2236" Action="Error" />
    <Rule Id="CA2237" Action="Error" />
    <Rule Id="CA2238" Action="Error" />
    <Rule Id="CA2239" Action="Error" />
    <Rule Id="CA2240" Action="Error" />
    <Rule Id="CA2241" Action="Error" />
    <Rule Id="CA2242" Action="Error" />
    <Rule Id="CA2243" Action="Error" />
  </Rules>
  <Rules AnalyzerId="Microsoft.CodeQuality.Analyzers" RuleNamespace="Microsoft.CodeQuality.Analyzers">
    <Rule Id="CA1062" Action="None" />
    <Rule Id="CA1707" Action="None" />
    <Rule Id="CA1715" Action="None" />
  </Rules>
  <Rules AnalyzerId="Microsoft.NetCore.Analyzers" RuleNamespace="Microsoft.NetCore.Analyzers">
    <Rule Id="CA1305" Action="None" />
    <Rule Id="CA1827" Action="None" />
    <Rule Id="CA1829" Action="None" />
    <Rule Id="CA1830" Action="None" />
    <Rule Id="CA1834" Action="None" />
    <Rule Id="CA2213" Action="None" />
  </Rules>
  <Rules AnalyzerId="Microsoft.NetCore.CSharp.Analyzers" RuleNamespace="Microsoft.NetCore.CSharp.Analyzers">
    <Rule Id="CA2016" Action="None" />
  </Rules>
  <Rules AnalyzerId="StyleCop.Analyzers" RuleNamespace="StyleCop.Analyzers">
    <Rule Id="SA0000" Action="Error" />
    <Rule Id="SA1025" Action="Error" />
    <Rule Id="SA1201" Action="Error" />
    <Rule Id="SA1202" Action="Error" />
    <Rule Id="SA1204" Action="Error" />
    <Rule Id="SA1208" Action="Error" />
    <Rule Id="SA1210" Action="Error" />
    <Rule Id="SA1400" Action="Error" />
    <Rule Id="SA1503" Action="Error" />
    <Rule Id="SA1505" Action="Error" />
    <Rule Id="SX1309" Action="Error" />
    <Rule Id="SA0001" Action="None" />
    <Rule Id="SA1000" Action="None" />
    <Rule Id="SA1001" Action="None" />
    <Rule Id="SA1002" Action="None" />
    <Rule Id="SA1003" Action="None" />
    <Rule Id="SA1004" Action="None" />
    <Rule Id="SA1005" Action="None" />
    <Rule Id="SA1006" Action="None" />
    <Rule Id="SA1007" Action="None" />
    <Rule Id="SA1008" Action="None" />
    <Rule Id="SA1009" Action="None" />
    <Rule Id="SA1010" Action="None" />
    <Rule Id="SA1011" Action="None" />
    <Rule Id="SA1012" Action="None" />
    <Rule Id="SA1013" Action="None" />
    <Rule Id="SA1014" Action="None" />
    <Rule Id="SA1015" Action="None" />
    <Rule Id="SA1016" Action="None" />
    <Rule Id="SA1017" Action="None" />
    <Rule Id="SA1018" Action="None" />
    <Rule Id="SA1019" Action="None" />
    <Rule Id="SA1020" Action="None" />
    <Rule Id="SA1021" Action="None" />
    <Rule Id="SA1022" Action="None" />
    <Rule Id="SA1023" Action="None" />
    <Rule Id="SA1024" Action="None" />
    <Rule Id="SA1026" Action="None" />
    <Rule Id="SA1027" Action="Error" />
    <Rule Id="SA1028" Action="None" />
    <Rule Id="SA1100" Action="None" />
    <Rule Id="SA1101" Action="None" />
    <Rule Id="SA1102" Action="None" />
    <Rule Id="SA1103" Action="None" />
    <Rule Id="SA1104" Action="None" />
    <Rule Id="SA1105" Action="None" />
    <Rule Id="SA1106" Action="Error" />
    <Rule Id="SA1107" Action="None" />
    <Rule Id="SA1108" Action="None" />
    <Rule Id="SA1110" Action="None" />
    <Rule Id="SA1111" Action="None" />
    <Rule Id="SA1112" Action="None" />
    <Rule Id="SA1113" Action="None" />
    <Rule Id="SA1114" Action="None" />
    <Rule Id="SA1115" Action="None" />
    <Rule Id="SA1116" Action="None" />
    <Rule Id="SA1117" Action="None" />
    <Rule Id="SA1118" Action="None" />
    <Rule Id="SA1119" Action="None" />
    <Rule Id="SA1120" Action="None" />
    <Rule Id="SA1121" Action="None" />
    <Rule Id="SA1122" Action="None" />
    <Rule Id="SA1123" Action="None" />
    <Rule Id="SA1124" Action="None" />
    <Rule Id="SA1125" Action="Error" />
    <Rule Id="SA1127" Action="None" />
    <Rule Id="SA1128" Action="None" />
    <Rule Id="SA1129" Action="None" />
    <Rule Id="SA1130" Action="Error" />
    <Rule Id="SA1131" Action="None" />
    <Rule Id="SA1132" Action="None" />
    <Rule Id="SA1133" Action="None" />
    <Rule Id="SA1134" Action="None" />
    <Rule Id="SA1137" Action="None" />
    <Rule Id="SA1200" Action="None" />
    <Rule Id="SA1203" Action="None" />
    <Rule Id="SA1205" Action="None" />
    <Rule Id="SA1206" Action="Error" />
    <Rule Id="SA1207" Action="None" />
    <Rule Id="SA1209" Action="None" />
    <Rule Id="SA1211" Action="None" />
    <Rule Id="SA1212" Action="None" />
    <Rule Id="SA1213" Action="None" />
    <Rule Id="SA1214" Action="None" />
    <Rule Id="SA1216" Action="None" />
    <Rule Id="SA1217" Action="None" />
    <Rule Id="SA1300" Action="Error" />
    <Rule Id="SA1302" Action="Error" />
    <Rule Id="SA1303" Action="None" />
    <Rule Id="SA1304" Action="None" />
    <Rule Id="SA1306" Action="Error" />
    <Rule Id="SA1307" Action="None" />
    <Rule Id="SA1308" Action="None" />
    <Rule Id="SA1309" Action="None" />
    <Rule Id="SA1310" Action="None" />
    <Rule Id="SA1311" Action="None" />
    <Rule Id="SA1312" Action="Error" />
    <Rule Id="SA1313" Action="Error" />
    <Rule Id="SA1314" Action="None" />
    <Rule Id="SA1401" Action="None" />
    <Rule Id="SA1402" Action="None" />
    <Rule Id="SA1403" Action="None" />
    <Rule Id="SA1404" Action="Error" />
    <Rule Id="SA1405" Action="None" />
    <Rule Id="SA1406" Action="None" />
    <Rule Id="SA1407" Action="None" />
    <Rule Id="SA1408" Action="None" />
    <Rule Id="SA1410" Action="None" />
    <Rule Id="SA1411" Action="None" />
    <Rule Id="SA1413" Action="None" />
    <Rule Id="SA1500" Action="Error" />
    <Rule Id="SA1501" Action="None" />
    <Rule Id="SA1502" Action="None" />
    <Rule Id="SA1504" Action="None" />
    <Rule Id="SA1506" Action="None" />
    <Rule Id="SA1507" Action="None" />
    <Rule Id="SA1508" Action="None" />
    <Rule Id="SA1509" Action="Error" />
    <Rule Id="SA1510" Action="None" />
    <Rule Id="SA1511" Action="None" />
    <Rule Id="SA1512" Action="None" />
    <Rule Id="SA1513" Action="None" />
    <Rule Id="SA1514" Action="None" />
    <Rule Id="SA1515" Action="None" />
    <Rule Id="SA1516" Action="None" />
    <Rule Id="SA1517" Action="None" />
    <Rule Id="SA1518" Action="None" />
    <Rule Id="SA1519" Action="Error" />
    <Rule Id="SA1520" Action="Error" />
    <Rule Id="SA1600" Action="None" />
    <Rule Id="SA1601" Action="None" />
    <Rule Id="SA1602" Action="None" />
    <Rule Id="SA1604" Action="None" />
    <Rule Id="SA1605" Action="None" />
    <Rule Id="SA1606" Action="None" />
    <Rule Id="SA1607" Action="None" />
    <Rule Id="SA1608" Action="None" />
    <Rule Id="SA1610" Action="None" />
    <Rule Id="SA1611" Action="None" />
    <Rule Id="SA1612" Action="None" />
    <Rule Id="SA1613" Action="None" />
    <Rule Id="SA1614" Action="None" />
    <Rule Id="SA1615" Action="None" />
    <Rule Id="SA1616" Action="None" />
    <Rule Id="SA1617" Action="None" />
    <Rule Id="SA1618" Action="None" />
    <Rule Id="SA1619" Action="None" />
    <Rule Id="SA1620" Action="None" />
    <Rule Id="SA1621" Action="None" />
    <Rule Id="SA1622" Action="None" />
    <Rule Id="SA1623" Action="None" />
    <Rule Id="SA1624" Action="None" />
    <Rule Id="SA1625" Action="None" />
    <Rule Id="SA1626" Action="None" />
    <Rule Id="SA1627" Action="None" />
    <Rule Id="SA1633" Action="None" />
    <Rule Id="SA1634" Action="None" />
    <Rule Id="SA1635" Action="None" />
    <Rule Id="SA1636" Action="None" />
    <Rule Id="SA1637" Action="None" />
    <Rule Id="SA1638" Action="None" />
    <Rule Id="SA1640" Action="None" />
    <Rule Id="SA1641" Action="None" />
    <Rule Id="SA1642" Action="None" />
    <Rule Id="SA1643" Action="None" />
    <Rule Id="SA1648" Action="None" />
    <Rule Id="SA1649" Action="Error" />
    <Rule Id="SA1651" Action="None" />
    <Rule Id="SA1652" Action="None" />
  </Rules>
</RuleSet>
```

**Comment by twilsman** [comment]
These are good suggestions.  I see no reason to not make these changes.

**Comment by bbarber** [comment]
Resolved in #12
